server/controllers/booking/booking.controller.js

```diff
@@ -42,11 +42,14 @@ const createBooking = function(req, res) {
             newBooking.user = selecteduser._id;
             newBooking.unlockCode = codeGenerator.generate();
             newBooking.disabled = false;
+            var hours = moment.duration(moment(endAt, 'YYYY/MM/DD HH:mm')
+            .diff(moment(startAt, 'YYYY/MM/DD HH:mm'))).asHours();
+            newBooking.totalCost =  hours*vehicle.vehicleType.hourlyRate;
             let alreadyBooked = false;
 
             var validateErrs = newBooking.validateSync();
             if (validateErrs) {
-              res.status(500).send(errs);
+              res.status(500).send(validateErrs);
             } else {
               Booking.find({ car: req.body.car, disabled: false }).exec(
```

Comment:
we could get rid of some of the checks if we said Booking.find( { car: req.body.car, disabled: false, endsAt: { $gt: startAt } } )
we could probably get even funkier with
`Booking.find({
  $and: {
    {car: req.body.car, disabled: false},
    $or: {
      $and: {$lt: { startAt: startAt }, $lt: { endAt: startAt }},
      $and: {$gt: { startAt: endAt }, $gt: { endAt: endAt }},
    }
  }
})` which is unfortunately mongoose's way of doing (car = req.body.car AND disabled = false) AND ( (startAt < startAt AND endAt < startAt) OR (startAt > endAt AND endAt > endAt))